e2e/e2e-scenario.sh

```diff
@@ -0,0 +1,192 @@
+    fi
+    set -x
+    echo "debug done"
+}
+
+KUBECONFIG=$(pwd)/kubeconfig
+export KUBECONFIG
+kubectl rollout status deploy/debug
+
+echo "Scenario is $SCENARIO_NAME"
+jq -s '.[0] * .[1]' nodebootstrapping_config.json scenarios/$SCENARIO_NAME/property-$SCENARIO_NAME.json > scenarios/$SCENARIO_NAME/nbc-$SCENARIO_NAME.json
+
+go test -run TestE2EBasic
+
+set +x
+if [ ! -f ~/.ssh/id_rsa ]; then
+    ssh-keygen -t rsa -b 4096 -f ~/.ssh/id_rsa -N ""
+fi
+set -x
+
+msiResourceID=$(jq -r '.identityProfile.kubeletidentity.resourceId' < cluster_info.json)
+MC_RESOURCE_GROUP_NAME="MC_${RESOURCE_GROUP_NAME}_${CLUSTER_NAME}_eastus"
+
+VMSS_NAME="$(mktemp -u abtest-XXXXXXX | tr '[:upper:]' '[:lower:]')"
+tee $SCENARIO_NAME-vmss.json > /dev/null <<EOF
+{
+    "group": "${MC_RESOURCE_GROUP_NAME}",
+    "vmss": "${VMSS_NAME}"
+}
+EOF
+
+cat $SCENARIO_NAME-vmss.json
+
+# Create a test VMSS with 1 instance 
+# TODO 3: Discuss about the --image version, probably go with aks-ubuntu-1804-gen2-2021-q2:latest
+#       However, how to incorporate chaning quarters?
+log "Creating VMSS"
+vmssStartTime=$(date +%s)
+az vmss create -n ${VMSS_NAME} \
+    -g $MC_RESOURCE_GROUP_NAME \
+    --admin-username azureuser \
+    --custom-data scenarios/$SCENARIO_NAME/$SCENARIO_NAME-cloud-init.txt \
+    --lb kubernetes --backend-pool-name aksOutboundBackendPool \
+    --vm-sku $VM_SKU \
+    --instance-count 1 \
+    --assign-identity $msiResourceID \
+    --image "microsoft-aks:aks:aks-ubuntu-1804-2022-q1:2022.02.01" \
+    --upgrade-policy-mode Automatic \
+    --ssh-key-values ~/.ssh/id_rsa.pub \
+    -ojson
+
+vmssEndTime=$(date +%s)
+log "Created VMSS in $((vmssEndTime-vmssStartTime)) seconds"
+
+# Get the name of the VM instance to later check with kubectl get nodes
+vmInstanceName=$(az vmss list-instances \
+                -n ${VMSS_NAME} \
+                -g $MC_RESOURCE_GROUP_NAME \
+                -ojson | \
+                jq -r '.[].osProfile.computerName'
+            )
+export vmInstanceName
+
+# Generate the extension from csecmd
+jq -Rs '{commandToExecute: . }' scenarios/$SCENARIO_NAME/$SCENARIO_NAME-cseCmd > scenarios/$SCENARIO_NAME/$SCENARIO_NAME-settings.json
+
+# Apply extension to the VM
+log "Applying extensions to VMSS"
+vmssExtStartTime=$(date +%s)
+set +e
+az vmss extension set --resource-group $MC_RESOURCE_GROUP_NAME \
+    --name CustomScript \
+    --vmss-name ${VMSS_NAME} \
+    --publisher Microsoft.Azure.Extensions \
+    --protected-settings scenarios/$SCENARIO_NAME/$SCENARIO_NAME-settings.json \
+    --version 2.0 \
+    -ojson
+retval=$?
+set -e
+
+vmssExtEndTime=$(date +%s)
+log "Applied extensions in $((vmssExtEndTime-vmssExtStartTime)) seconds"
+
+FAILED=0
+# Check if the node joined the cluster
+if [[ "$retval" != "0" ]]; then
+    err "cse failed to apply"
+    debug
+    tail -n 50 $SCENARIO_NAME-logs/cluster-provision.log || true
+    exit 1
+fi
+
+# Sleep to let the automatic upgrade of the VM finish
+waitForNodeStartTime=$(date +%s)
```

Comment:
this script is getting a bit long, wonder if we can refactor to tidy up a bit...